task/host-sp-comms/src/main.rs

```diff
@@ -843,8 +851,36 @@ impl ServerImpl {
                 Some(SpToHost::Alert { action: 0 })
             }
             HostToSp::RotRequest => {
-                // TODO forward request to RoT
-                Some(SpToHost::RotResponse)
+                match attest_data::messages::parse_message(data) {
+                    Ok((command, data)) => {
+                        let n = usize::min(MAX_DATA_LEN, data.len());
+
+                        let mut data_buf: [u8; MAX_DATA_LEN] =
+                            [0; MAX_DATA_LEN];
+                        if n > 0 {
+                            data_buf[..n].copy_from_slice(&data[..n]);
+                        }
+
+                        self.handle_sprot(header.sequence, command, &data_buf);
```

Comment:
Just making sure I understand - we pass `&data_buf` here instead of `&data[..n]` because attestation expects the data to always be `MAX_DATA_LEN` and zero padded if needed?